src/imitation/data/datasets.py

```diff
@@ -133,32 +133,35 @@ def sample(self, n_samples: int) -> Dict[str, np.ndarray]:
         return {k: v[inds] for k, v in self.data_map.items()}
 
 
-class TransitionsDictDatasetAdaptor(Dataset[types.Transitions]):
+S = TypeVar("S", bound=types.TransitionsMinimal)
+
+
+class TransitionsDictDatasetAdaptor(Dataset[S]):
     def __init__(
         self,
-        transitions: types.Transitions,
+        transitions: S,
         dict_dataset_cls: Type[DictDataset] = RandomDictDataset,
         dict_dataset_cls_kwargs: Optional[Mapping] = None,
     ):
         """Adapts a `DictDataset` class to build `Transitions` `Dataset`.
 
         Args:
-            transitions: An `Transitions` instance to be sampled from and stored
+            transitions: An `TransitionsMinimal` instance to be sampled from and stored
                 internally as a `dict` of copied arrays. Note that `sample` will return
-                the same type of `Transitions` as the type of this arg. In other words,
-                if `transitions` is an instance of `TransitionsWithRew`, then
-                `.sample()` also returns `TransitionsWithRew`.
+                the same subclass type of `TransitionsMinimal` as the type of this arg.
             dict_dataset_cls: `DictDataset` class to be adapted.
             dict_dataset_kwargs: Optional kwargs for initializing `DictDataset` class.
         """
         data_map: Dict[str, np.ndarray] = dataclasses.asdict(transitions)
+        data_map["infos"] = np.array(data_map["infos"])
```

Comment:
Why do we need this? Isn't it already an array?

Reply:
Unnecesary now, thanks for catching. This was leftover back from when `infos` could be a sequence, and I'd get an error from `infos[perm_idx]` inside `EpochOrderDictDataset`.